src/rgw/rgw_sal_motr.cc

```diff
@@ -1844,12 +1844,14 @@ int MotrObject::write_mobj(const DoutPrefixProvider *dpp, bufferlist&& data, uin
 int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t end, RGWGetDataCB* cb)
 {
   int rc;
-  unsigned bs, actual, left;
+  unsigned bs, actual, left, start, bloff, block_start_off;
   struct m0_op *op;
   struct m0_bufvec buf;
   struct m0_bufvec attr;
   struct m0_indexvec ext;
 
+  start = off;
+  off = 0;
```

Comment:
We need to check and confirm the side effects of making `off=0` before calculating the bs value. There should not be any major issue but it's better to have an idea.

Reply:
For normal get object call off will come 0 only that is my understanding on this, in byte range case this will come different but that value we are storing in start and making off to 0, so that bs can be calculated and we can fetch the first block from 0th offset.

Reply:
seems alright